File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {

	input = input.trim();

	if (isYelling(input)) {
		return 'Whoa, chill out!';
	}
	else if (isQuestion(input)) {
		return 'Sure.';
	}
	else if (isSilent(input)) {
		return 'Fine. Be that way!';
	}
	else {
		return 'Whatever.';
	}
};

function isYelling(input) {
	return input.toUpperCase() === input 
		&& input.match('[a-zA-z]+');
}

function isQuestion(input) {
	return input.slice(-1) === '?';
}

function isSilent(input) {
	return input === '';
}

module.exports = Bob;

File: fsharp/bob/Bob.fs
module Bob
open System
open System.Text.RegularExpressions

type Bob(sentence:string) = 

    let isYelling() = 
        String.Compare(sentence.ToUpper(), sentence) = 0 
            && Regex.IsMatch(sentence, @"[A-Za-z]")
    let isQuestion() = sentence.EndsWith("?")
    let isSilent() = String.IsNullOrEmpty(sentence.Trim())

    member this.hey() =
        if isYelling() then "Whoa, chill out!"
        elif isQuestion() then "Sure."
        elif isSilent() then "Fine. Be that way!"
        else "Whatever."
